package.json

```
@@ -25,12 +25,12 @@
     "test:watch": "jest --watchAll"
   },
   "dependencies": {
-    "@ethersproject/bignumber": "^5.7.0",
     "@ethersproject/bytes": "^5.7.0",
-    "@ethersproject/providers": "^5.7.0",
     "@metamask/base-controller": "^3.2.1",
     "@metamask/controller-utils": "^5.0.0",
+    "@metamask/eth-query": "^3.0.1",
```

```suggestion
    "@metamask/base-controller": "^4.0.0",
    "@metamask/controller-utils": "^6.1.0",
    "@metamask/eth-query": "^4.0.0",
```

With these version bumps (and https://github.com/MetaMask/smart-transactions-controller/pull/237#discussion_r1402649548), all `// @ts-expect error TODO: Provider type alignment` directives can be removed.